src/tinypeel/Peeling/PeelingIO.py

```diff
@@ -76,30 +76,39 @@ def writeGenotypes(pedigree, genoProbFunc):
     args = InputOutput.args
     if not args.no_dosage:
         writeDosages(pedigree, genoProbFunc, args.out + ".dosage.txt")
-    if args.haps:
-        writeGenoProbs(pedigree, genoProbFunc, args.out + ".haps")
-
-    if args.calling_threshold is not None:
-        for thresh in args.calling_threshold:
+    if args.phased_geno_prob:
+        writePhasedGenoProbs(pedigree, genoProbFunc, args.out + ".phased_geno_prob.txt")
+    if args.geno_prob:
+        writeGenoProbs(pedigree, genoProbFunc, args.out + ".geno_prob.txt")
+    if args.geno_threshold is not None:
+        for thresh in args.geno_threshold:
+            if thresh < 1 / 3:
+                thresh = 1 / 3
             if args.binary_call_files:
                 writeBinaryCalledGenotypes(
                     pedigree, genoProbFunc, args.out + ".called." + str(thresh), thresh
                 )
-            if not args.binary_call_files:
+            if not args.binary_call_files and args.geno:
```

Comment:
@AprilYUZhang probably best to remove `and args.geno` here (I appreciate that @XingerTang said to do it), and use it above in changing `if args.geno_threshold is not None:` to `if args.geno and args.geno_threshold is not None:`

Reply:
I agree with @gregorgorjanc, but I noticed another place could also be improved at the same time. I think it might be better to change from ``if args.geno and args.geno_threshold is not None:`` to ``if args.geno and args.geno_threshold:`` . 

I wrote a small testing file for the arguments input that allows multiple float inputs. 

```testing.py
import argparse

parser = argparse.ArgumentParser(description = 'Test')
parser.add_argument('-arg1',
                    default=None,
                    required=False,
                    type=float,
                    nargs="*"
                    )
args = vars(parser.parse_args())
if args["arg1"] is not None:
    print(args["arg1"])
else:
    print("expected")
```
If we now run the following code
```
python /Users/evie/test/src/test/testing.py -arg1
```
Then it would give ``[]``, instead of notice that there is actually no input.

But if the code is as follows (I only change the if condition)
```testing.py
import argparse

parser = argparse.ArgumentParser(description = 'Test')
parser.add_argument('-arg1',
                    default=None,
                    required=False,
                    type=float,
                    nargs="*"
                    )
args = vars(parser.parse_args())
if args["arg1"]:
    print(args["arg1"])
else:
    print("expected")
```
then ``python /Users/evie/test/src/test/testing.py -arg1`` would gives ``expected``.

Reply:
@XingerTang @AprilYUZhang I will follow your advice here;)